{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "eba68a55-9ab1-4886-b1aa-4e5fe3f352db",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "b85a46cf-9f0c-408f-8bf9-88106160b439",
   "metadata": {},
   "outputs": [],
   "source": [
    "taxonomy = {}\n",
    "with open('./archaea_overview.txt') as fh:\n",
    "    header = fh.readline()\n",
    "    for line in fh:\n",
    "        line = line.strip().split('\\t')\n",
    "        genus = line[0].split(' ')[0]\n",
    "        taxonomy[genus] = line[1:4]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "13abdb2b-e0df-4ee1-b990-f44af7f9be63",
   "metadata": {},
   "outputs": [],
   "source": [
    "output = open('archaea_OrganismTable.tsv', 'w')\n",
    "with open('./refseq_assembly_summary_archaea.txt') as fh:\n",
    "    for line in fh:\n",
    "        if not line.startswith('#'):\n",
    "            line = line.strip().split('\\t')\n",
    "            organism = line[7]\n",
    "            genus = organism.split(' ')[0]\n",
    "            try:\n",
    "                a = taxonomy[genus]\n",
    "                print(*line, *a, sep='\\t', file=output)\n",
    "            except KeyError:\n",
    "                pass\n",
    "        elif line.startswith('# assembly'):\n",
    "            line = line.strip()[2:] + '\\t' + '\\t'.join(['Kingdom', 'Group', 'Subgroup'])\n",
    "            print(line, file=output)\n",
    "output.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "63957f69-a0cf-4688-85b2-45750ee6447f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "df = pd.read_csv('./archaea_OrganismTable.tsv', sep='\\t')\n",
    "df = df[['organism_name', 'Kingdom', 'Group', 'Subgroup', 'infraspecific_name', 'isolate', 'assembly_accession', 'assembly_level', 'asm_name', 'refseq_category', 'species_taxid', 'ftp_path']]\n",
    "# df.to_csv('./archaea_MainTable.tsv', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "26d44f9d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>organism_name</th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>assembly_level</th>\n",
       "      <th>asm_name</th>\n",
       "      <th>refseq_category</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Methanobacterium bryantii</td>\n",
       "      <td>GCF_002287175.1</td>\n",
       "      <td>Contig</td>\n",
       "      <td>ASM228717v1</td>\n",
       "      <td>representative genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Methanobacterium formicicum</td>\n",
       "      <td>GCF_000762265.1</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM76226v1</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Methanobacterium formicicum</td>\n",
       "      <td>GCF_001458655.1</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>Mb9</td>\n",
       "      <td>representative genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Methanobrevibacter smithii</td>\n",
       "      <td>GCF_000824705.1</td>\n",
       "      <td>Contig</td>\n",
       "      <td>Methanobrevibacter smithii</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Methanobrevibacter smithii</td>\n",
       "      <td>GCF_002252585.1</td>\n",
       "      <td>Contig</td>\n",
       "      <td>ASM225258v1</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1204</th>\n",
       "      <td>Halobacterium sp. Gai3-17</td>\n",
       "      <td>GCF_021249345.1</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM2124934v1</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1205</th>\n",
       "      <td>Halobacterium sp. XZYJT26</td>\n",
       "      <td>GCF_021249405.1</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM2124940v1</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1206</th>\n",
       "      <td>Halobacterium sp. TGN-42-S1</td>\n",
       "      <td>GCF_021729015.1</td>\n",
       "      <td>Contig</td>\n",
       "      <td>ASM2172901v1</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1207</th>\n",
       "      <td>Halobaculum sp. CBA1158</td>\n",
       "      <td>GCF_021431925.1</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM2143192v1</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1208</th>\n",
       "      <td>Methanothermobacter sp. K4</td>\n",
       "      <td>GCF_022014235.1</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>ASM2201423v1</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1209 rows Ã— 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    organism_name assembly_accession   assembly_level  \\\n",
       "0       Methanobacterium bryantii    GCF_002287175.1           Contig   \n",
       "1     Methanobacterium formicicum    GCF_000762265.1  Complete Genome   \n",
       "2     Methanobacterium formicicum    GCF_001458655.1  Complete Genome   \n",
       "3      Methanobrevibacter smithii    GCF_000824705.1           Contig   \n",
       "4      Methanobrevibacter smithii    GCF_002252585.1           Contig   \n",
       "...                           ...                ...              ...   \n",
       "1204    Halobacterium sp. Gai3-17    GCF_021249345.1  Complete Genome   \n",
       "1205    Halobacterium sp. XZYJT26    GCF_021249405.1  Complete Genome   \n",
       "1206  Halobacterium sp. TGN-42-S1    GCF_021729015.1           Contig   \n",
       "1207      Halobaculum sp. CBA1158    GCF_021431925.1  Complete Genome   \n",
       "1208   Methanothermobacter sp. K4    GCF_022014235.1         Scaffold   \n",
       "\n",
       "                        asm_name        refseq_category  \n",
       "0                    ASM228717v1  representative genome  \n",
       "1                     ASM76226v1                     na  \n",
       "2                            Mb9  representative genome  \n",
       "3     Methanobrevibacter smithii                     na  \n",
       "4                    ASM225258v1                     na  \n",
       "...                          ...                    ...  \n",
       "1204                ASM2124934v1                     na  \n",
       "1205                ASM2124940v1                     na  \n",
       "1206                ASM2172901v1                     na  \n",
       "1207                ASM2143192v1                     na  \n",
       "1208                ASM2201423v1                     na  \n",
       "\n",
       "[1209 rows x 5 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = df[['organism_name','assembly_accession', 'assembly_level','asm_name','refseq_category']]\n",
    "\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "84ec54da-66e6-4f2d-bc99-b4b8facfb481",
   "metadata": {},
   "outputs": [],
   "source": [
    "# def get_genomeinfo(fname):\n",
    "#     gsize = 0\n",
    "#     GC = 0\n",
    "#     with open(f'./genomes/{fname}_genomic.fna') as fh:\n",
    "#         for line in fh:\n",
    "#             if not line.startswith('>'):\n",
    "#                 line = line.strip()\n",
    "#                 for c in line:\n",
    "#                     c = c.upper()\n",
    "#                     gsize += 1\n",
    "#                     if c=='C' or c=='G': GC += 1\n",
    "#     return [gsize, round((GC/gsize)*100, 2)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "505f4162-6833-42af-9bf7-08512e163d30",
   "metadata": {},
   "outputs": [],
   "source": [
    "# def get_repeatinfo(fname):\n",
    "#     total_repeats = 0\n",
    "#     total_repeat_bases = 0\n",
    "#     with open(f'./imperf/{fname}_genomic_imperf.tsv') as fh:\n",
    "#         for line in fh:\n",
    "#             line = line.strip().split('\\t')\n",
    "#             total_repeats += 1\n",
    "#             total_repeat_bases += int(line[3])\n",
    "#     return [total_repeats, total_repeat_bases]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "e516629e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# PERF ARCHAEA\n",
    "\n",
    "# Step1 Data Cleanup for the \"Perf repeats\"\n",
    "\n",
    "def get_perf_genomeinfo(fname):\n",
    "    gsize = 0\n",
    "    GC = 0\n",
    "    with open(f'./genomes/{fname}_genomic.fna') as fh:\n",
    "        for line in fh:\n",
    "            if not line.startswith('>'):\n",
    "                line = line.strip()\n",
    "                for c in line:\n",
    "                    c = c.upper()\n",
    "                    gsize += 1\n",
    "                    if c=='C' or c=='G': GC += 1\n",
    "    return [gsize, round((GC/gsize)*100, 2)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "7ac6c825",
   "metadata": {},
   "outputs": [],
   "source": [
    "# PERF ARCHAEA\n",
    "\n",
    "# Step1 Data Cleanup for the \"Perf\" repeats\n",
    "\n",
    "def get_perf_repeatinfo(fname):\n",
    "    total_perf_repeats = 0\n",
    "    total_perf_repeats_bases = 0\n",
    "    with open(f'./genomes/{fname}_genomic_divissr.tsv') as fh:\n",
    "        for line in fh:\n",
    "            line = line.strip().split('\\t')\n",
    "            total_perf_repeats += 1\n",
    "            total_perf_repeats_bases += int(line[4])\n",
    "    return [total_perf_repeats, total_perf_repeats_bases]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "d4679ae3-0760-4d44-b439-e9a2902753ad",
   "metadata": {},
   "outputs": [],
   "source": [
    "# PERF ARCHAEA\n",
    "\n",
    "output = open('./archaea_Perf_MainTableInfo.tsv', 'w')\n",
    "with open('./archaea_MainTable.tsv') as fh:\n",
    "    header = fh.readline().strip()\n",
    "    print(header, 'genome_size', 'gc_percent', 'total_repeats', 'total_repeat_bases', 'total_repeats_norm', 'total_repeat_bases_norm', sep='\\t', file=output)\n",
    "    for line in fh:\n",
    "        line = line.strip().split('\\t')\n",
    "        file_name = line[-1]\n",
    "        ginfo = get_perf_genomeinfo(file_name)\n",
    "        rep_info = get_perf_repeatinfo(file_name)\n",
    "        line += ginfo\n",
    "        line += rep_info\n",
    "        line += [round((rep_info[0]/ginfo[0])*1000, 5), round((rep_info[1]/ginfo[0])*1000, 5)]\n",
    "        print(*line, sep='\\t', file=output)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "08896884",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>organism_name</th>\n",
       "      <th>Kingdom</th>\n",
       "      <th>Group</th>\n",
       "      <th>Subgroup</th>\n",
       "      <th>infraspecific_name</th>\n",
       "      <th>isolate</th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>assembly_level</th>\n",
       "      <th>asm_name</th>\n",
       "      <th>refseq_category</th>\n",
       "      <th>species_taxid</th>\n",
       "      <th>file_name</th>\n",
       "      <th>genome_size</th>\n",
       "      <th>gc_percent</th>\n",
       "      <th>total_repeats</th>\n",
       "      <th>total_repeat_bases</th>\n",
       "      <th>total_repeats_norm</th>\n",
       "      <th>total_repeat_bases_norm</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Methanobacterium bryantii</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Methanomada group</td>\n",
       "      <td>strain=M.o.H.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCF_002287175.1</td>\n",
       "      <td>Contig</td>\n",
       "      <td>ASM228717v1</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>2161.0</td>\n",
       "      <td>GCF_002287175.1_ASM228717v1</td>\n",
       "      <td>3466370.0</td>\n",
       "      <td>33.19</td>\n",
       "      <td>2860.0</td>\n",
       "      <td>35950.0</td>\n",
       "      <td>0.82507</td>\n",
       "      <td>10.37108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Methanobacterium formicicum</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Methanomada group</td>\n",
       "      <td>strain=BRM9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCF_000762265.1</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>ASM76226v1</td>\n",
       "      <td>na</td>\n",
       "      <td>2162.0</td>\n",
       "      <td>GCF_000762265.1_ASM76226v1</td>\n",
       "      <td>2449987.0</td>\n",
       "      <td>41.34</td>\n",
       "      <td>1669.0</td>\n",
       "      <td>20975.0</td>\n",
       "      <td>0.68123</td>\n",
       "      <td>8.56127</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Methanobacterium formicicum</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Methanomada group</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mb9</td>\n",
       "      <td>GCF_001458655.1</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>Mb9</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>2162.0</td>\n",
       "      <td>GCF_001458655.1_Mb9</td>\n",
       "      <td>2494510.0</td>\n",
       "      <td>41.14</td>\n",
       "      <td>1710.0</td>\n",
       "      <td>21513.0</td>\n",
       "      <td>0.68551</td>\n",
       "      <td>8.62414</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Methanobrevibacter smithii</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Methanomada group</td>\n",
       "      <td>strain=ACE6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCF_000824705.1</td>\n",
       "      <td>Contig</td>\n",
       "      <td>Methanobrevibacter smithii</td>\n",
       "      <td>na</td>\n",
       "      <td>2173.0</td>\n",
       "      <td>GCF_000824705.1_Methanobrevibacter_smithii</td>\n",
       "      <td>1718384.0</td>\n",
       "      <td>31.51</td>\n",
       "      <td>1595.0</td>\n",
       "      <td>20027.0</td>\n",
       "      <td>0.92820</td>\n",
       "      <td>11.65455</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Methanobrevibacter smithii</td>\n",
       "      <td>Archaea</td>\n",
       "      <td>Euryarchaeota</td>\n",
       "      <td>Methanomada group</td>\n",
       "      <td>strain=WWM1085</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCF_002252585.1</td>\n",
       "      <td>Contig</td>\n",
       "      <td>ASM225258v1</td>\n",
       "      <td>na</td>\n",
       "      <td>2173.0</td>\n",
       "      <td>GCF_002252585.1_ASM225258v1</td>\n",
       "      <td>1926345.0</td>\n",
       "      <td>30.29</td>\n",
       "      <td>1847.0</td>\n",
       "      <td>23326.0</td>\n",
       "      <td>0.95881</td>\n",
       "      <td>12.10894</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 organism_name  Kingdom          Group           Subgroup  \\\n",
       "0    Methanobacterium bryantii  Archaea  Euryarchaeota  Methanomada group   \n",
       "1  Methanobacterium formicicum  Archaea  Euryarchaeota  Methanomada group   \n",
       "2  Methanobacterium formicicum  Archaea  Euryarchaeota  Methanomada group   \n",
       "3   Methanobrevibacter smithii  Archaea  Euryarchaeota  Methanomada group   \n",
       "4   Methanobrevibacter smithii  Archaea  Euryarchaeota  Methanomada group   \n",
       "\n",
       "  infraspecific_name isolate assembly_accession   assembly_level  \\\n",
       "0      strain=M.o.H.     NaN    GCF_002287175.1           Contig   \n",
       "1        strain=BRM9     NaN    GCF_000762265.1  Complete Genome   \n",
       "2                NaN     Mb9    GCF_001458655.1  Complete Genome   \n",
       "3        strain=ACE6     NaN    GCF_000824705.1           Contig   \n",
       "4     strain=WWM1085     NaN    GCF_002252585.1           Contig   \n",
       "\n",
       "                     asm_name        refseq_category  species_taxid  \\\n",
       "0                 ASM228717v1  representative genome         2161.0   \n",
       "1                  ASM76226v1                     na         2162.0   \n",
       "2                         Mb9  representative genome         2162.0   \n",
       "3  Methanobrevibacter smithii                     na         2173.0   \n",
       "4                 ASM225258v1                     na         2173.0   \n",
       "\n",
       "                                    file_name  genome_size  gc_percent  \\\n",
       "0                 GCF_002287175.1_ASM228717v1    3466370.0       33.19   \n",
       "1                  GCF_000762265.1_ASM76226v1    2449987.0       41.34   \n",
       "2                         GCF_001458655.1_Mb9    2494510.0       41.14   \n",
       "3  GCF_000824705.1_Methanobrevibacter_smithii    1718384.0       31.51   \n",
       "4                 GCF_002252585.1_ASM225258v1    1926345.0       30.29   \n",
       "\n",
       "   total_repeats  total_repeat_bases  total_repeats_norm  \\\n",
       "0         2860.0             35950.0             0.82507   \n",
       "1         1669.0             20975.0             0.68123   \n",
       "2         1710.0             21513.0             0.68551   \n",
       "3         1595.0             20027.0             0.92820   \n",
       "4         1847.0             23326.0             0.95881   \n",
       "\n",
       "   total_repeat_bases_norm  \n",
       "0                 10.37108  \n",
       "1                  8.56127  \n",
       "2                  8.62414  \n",
       "3                 11.65455  \n",
       "4                 12.10894  "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# PERF_ARCHAEA\n",
    "\n",
    "import pandas as pd\n",
    "\n",
    "df = pd.read_csv('./archaea_Perf_MainTableInfo.tsv', sep = \"\\t\")\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4cd04bc5",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
